samza-core/src/main/java/org/apache/samza/coordinator/MetadataResourceUtil.java

```diff
@@ -0,0 +1,84 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.coordinator;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.samza.checkpoint.CheckpointManager;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.TaskConfigJava;
+import org.apache.samza.job.model.JobModel;
+import org.apache.samza.metadatastore.MetadataStore;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.storage.ChangelogStreamManager;
+
+
+/**
+ * Loads the managers responsible for the creation and loading of metadata related resources.
+ * TODO: Replace with a metadata admin interface when the {@link MetadataStore} is fully augmented to handle all metadata sources.
+ */
+public class MetadataResourceUtil {
+  private final Config config;
+  private final CheckpointManager checkpointManager;
+  private final JobModel jobModel; // TODO: Should be loaded by metadata store in the future
+
+  /**
+   * @param metadataStore the {@link MetadataStore} that manages the metadata. Assumption is that the metadataStore is
+   *                      initialized and not stopped.
+   * @param jobModel the loaded {@link JobModel}
+   * @param metricsRegistry the registry for reporting metrics.
+   * @param configOverride the config override to use instead of the one in the jobModel.
+   */
+  public MetadataResourceUtil(MetadataStore metadataStore, JobModel jobModel, MetricsRegistry metricsRegistry, Config configOverride) {
+    this.config = configOverride;
+    this.jobModel = jobModel;
+    this.checkpointManager = new TaskConfigJava(config).getCheckpointManager(metricsRegistry);
+  }
+
+  /**
+   * @param metadataStore the {@link MetadataStore} that manages the metadata
+   * @param jobModel the loaded {@link JobModel}
+   * @param metricsRegistry the registry for reporting metrics.
+   */
+  public MetadataResourceUtil(MetadataStore metadataStore, JobModel jobModel, MetricsRegistry metricsRegistry) {
+    this(metadataStore, jobModel, metricsRegistry, jobModel.getConfig());
+  }
+
+  @VisibleForTesting
+  MetadataResourceUtil(CheckpointManager checkpointManager, JobModel jobModel) {
+    this.config = jobModel.getConfig();
+    this.jobModel = jobModel;
+    this.checkpointManager = checkpointManager;
+  }
+
+  /**
+   * Creates and loads the required metadata resources for checkpoints, changelog stream and other
+   * resources related to the metadata system
+   */
+  public void createResources() {
```

Comment:
For the follow on startpoint work, I plan to add either the fan out call here for the fan out model or the ack clean up call here for the intent-ack model.